File: services/frontend/src/api/schemas.ts
export interface Appointment {
    id: string;
    patient_id: string;
    provider_id: string;
    start_ts: string;
    end_ts: string;
    status: string;
    visit_type: string;
    location_id: string;
}

export interface AvailabilitySlot {
    start: string;
    end: string;
}

export interface Availability {
    provider_id: string;
    from: string;
    to: string;
    slots: AvailabilitySlot[];
}

File: services/appointment_service/app/schemas.py
from pydantic import BaseModel
from datetime import datetime

class BookRequest(BaseModel):
    patient_id: str
    provider_id: str
    visit_type: str
    start_ts: datetime
    end_ts: datetime
    location_id: str

class BookResponse(BaseModel):
    appointment_id: str
    status: str
